src/KnownRelaysManager.ts

```diff
@@ -0,0 +1,292 @@
+      };
+    });
+
+    this.allRelayers = await this.getRelayDataForManagers(
+      recentlyActiveRelayManagers
+    );
+
+    log.debug('KnownRelaysManager - Get relay info for Managers done');
+    log.debug('KnownRelaysManager - Refresh done');
+  }
+
+  async getRelayDataForManagers(
+    relayManagers: Set<string>
+  ): Promise<IRelayHub.RelayManagerDataStruct[]> {
+    if (relayManagers.size === 0) {
+      return [];
+    }
+    const activeRelays = await this.contractInteractor.getActiveRelayInfo(
+      relayManagers
+    );
+    // As 'topics' are used as 'filter', having an empty set results in querying all register events.
+
+    return activeRelays.filter(this.relayFilter);
+  }
+
+  splitRange(
+    fromBlock: number,
+    toBlock: number,
+    splits: number
+  ): Array<{ fromBlock: number; toBlock: number }> {
+    const totalBlocks = toBlock - fromBlock + 1;
+    const splitSize = Math.ceil(totalBlocks / splits);
+
+    const ret: Array<{ fromBlock: number; toBlock: number }> = [];
+    let b;
+    for (b = fromBlock; b < toBlock; b += splitSize) {
+      ret.push({
+        fromBlock: b,
+        toBlock: Math.min(toBlock, b + splitSize - 1),
+      });
+    }
+
+    return ret;
+  }
+
+  // return events from hub. split requested range into "window parts", to avoid
+  // fetching too many events at once.
+  getPastEventsForHub(
+    fromBlock: number,
+    toBlock: number
+  ): Promise<TypedEvent[][]> {
+    const rangeParts = this.splitRange(
+      fromBlock,
+      toBlock,
+      this.relayLookupWindowParts
+    );
+
+    const pastEventsPromises = rangeParts.map(({ fromBlock, toBlock }) => {
+      return this.contractInteractor.getPastEventsForHub({
+        fromBlock,
+        toBlock,
+      });
+    });
+
+    return Promise.all(pastEventsPromises).then(
+      (pastEventArray: TypedEvent[][][]) => {
+        return pastEventArray.flat();
+      }
+    );
+  }
+
+  async _fetchRecentlyActiveRelayManagers(): Promise<Set<string>> {
+    const toBlock = await this.contractInteractor.getBlockNumber();
+    const fromBlock = Math.max(
+      0,
+      toBlock - this.config.relayLookupWindowBlocks
+    );
+
+    const relayEvents = await this.getPastEventsForHub(fromBlock, toBlock);
+
+    log.info(`fetchRelaysAdded: found ${relayEvents.length} events`);
+    const foundRelayManagers: Set<string> = new Set();
+    relayEvents.forEach((eventList: GenericEvent[]) => {
+      // TODO: remove relay managers who are not staked
+      // if (event.event === 'RelayRemoved') {
+      //   foundRelays.delete(event.returnValues.relay)
+      // } else {
+
+      const genericEvent = eventList.find((event) => event.args);
+      if (!genericEvent) {
+        log.info('No event with suitable args found');
+
+        return;
+      }
+
+      const {
+        args: { relayManager },
+      } = genericEvent;
+
+      if (!genericEvent) {
+        log.info('No relayManager found in event');
+
+        return;
+      }
```

Comment:
I think this may never happen, since we're doing the exact check at line 182. Am I missing something?

Reply:
You are right, it was an oversight from my part. Removed.